### PR TITLE
NSX-T Security Group resource and datasource

### PR DESCRIPTION
**Pull request by Didainius**
This PR introduces NSX-T Security Groups datasource and resource.

Additionally it:
* Fixes a few typos and adds an example in Routed Network V2 docs
* Adds compatibility with Terraform 0.15.0 for `TestAccVcdNsxtStandaloneVmTemplate` which complained about `sensitive` `output` 

On the code side:
* Adds VCDClient.GetNsxtEdgeGateway, VCDClient.GetNsxtEdgeGatewayById, and
  GetNsxtEdgeGatewayFromResourceById convenience functions
* Adds `importStateIdNsxtEdgeGatewayObject` function for import testing of NSX-T Edge Gateway child
  objects

* Also - adds a validator that `-v` flag is passed to tests as running it without it can cause panic in Terraform acceptance tests.

### CODE REVIEW

File: website/docs/r/nsxt_security_group.html.markdown [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+# vcd\_nsxt\_security\_group
+
+Provides a resource to manage NSX-T Security Group. Security groups are groups of data center group
+networks to which distributed firewall rules apply. Grouping networks helps you to reduce the total
+number of distributed firewall rules to be created.
+
+Supported in provider *v3.3+* and VCD 10.1+ with NSX-T backed VDCs.
+
+## Specific usage notes
```
Comment (lvirbalas) [inline website/docs/r/nsxt_security_group.html.markdown:19]: This section is empty.

File: website/docs/r/nsxt_security_group.html.markdown [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+## Example Usage 1 (Security Group with member networks)
+
+```hcl
+resource "vcd_nsxt_security_group" "frontend-servers" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  # refering to a datasource for existing NSX-T Edge Gateway
```
Comment (lvirbalas) [inline website/docs/r/nsxt_security_group.html.markdown:28]:
```suggestion
  # Referring to a datasource for existing NSX-T Edge Gateway
```

File: website/docs/r/nsxt_security_group.html.markdown [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+}
+```
+
+## Example Usage 2 (Empty Security Group)
+```hcl
+resource "vcd_nsxt_security_group" "group1" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  # refering to a datasource for existing NSX-T Edge Gateway
```
Comment (lvirbalas) [inline website/docs/r/nsxt_security_group.html.markdown:44]:
```suggestion
  # Referring to a datasource for existing NSX-T Edge Gateway
```

File: website/docs/r/nsxt_security_group.html.markdown [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+```hcl
+resource "vcd_nsxt_security_group" "frontend-servers" {
+  org  = "my-org"
+  vdc  = "my-nsxt-vdc"
+
+  # refering to a datasource for existing NSX-T Edge Gateway
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name = "frontend-servers"
+  description = "Security group consisting of frontend servers"
```
Comment (lvirbalas) [inline website/docs/r/nsxt_security_group.html.markdown:32]: Why it says frontend servers, while in practice it is linking to a network?

**Review by lvirbalas — CHANGES_REQUESTED** [review]

File: website/docs/r/nsxt_security_group.html.markdown (hunk 4 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown:32]: I can change, but security groups are quite often created "by purpose". So when you open a firewall rule list you see something similar to:
```
Open Port TCP/443 to "Frontend servers"
Open Port TCP/3306 from "Frontend servers" to "Database servers"
```

File: website/docs/r/nsxt_security_group.html.markdown (hunk 1 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown]: Indeed. Left it here as I moved the text to Member VMs section:
```
~> There may be cases where Org Networks and Security Groups are already created, but
not all VMs are already created and not shown in this structure. Additional `depends_on` can ensure
that Security Group is created only after all networks and VMs are there.

```

Fixed.

File: website/docs/r/nsxt_security_group.html.markdown (hunk 2 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown]: Thanks. Fixed.

File: website/docs/r/nsxt_security_group.html.markdown (hunk 3 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown]: Thanks. Fixed.

File: website/docs/r/nsxt_security_group.html.markdown (hunk 4 above)
Comment (lvirbalas) [reply website/docs/r/nsxt_security_group.html.markdown:30]: I'm just thinking about new users. Maybe amend description like this, so it's clear for everyone?
`description = "Security group for a network connecting the frontend servers"`

Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown]: No objections - changed.

**Review by lvirbalas — APPROVED** [review]

File: vcd/resource_vcd_nsxt_security_group.go [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+			Description: "Member VM ID",
+		},
+		"vm_name": {
+			Type:        schema.TypeString,
+			Computed:    true,
+			Description: "Member VM Name",
+		},
+		"vapp_id": {
+			Type:     schema.TypeString,
+			Optional: true, Computed: true,
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_security_group.go:87]:
```suggestion
			Optional: true,
			Computed: true,
```

File: vcd/resource_vcd_nsxt_security_group_test.go [hunk 6]
```diff
@@ -0,0 +1,477 @@
...
+}
+
+const testAccNsxtSecurityGroupPrereqs = testAccNsxtSecurityGroupPrereqsEmpty + `
+resource "vcd_network_routed_v2" "nsxt-backed" {
+	# This value could be larger to test more members, but left 2 for the sake of testing speed
+	count = 2
+
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+	name        = "nsxt-routed-${count.index}"
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_security_group_test.go:218]: This script is nor well formatted

File: vcd/resource_vcd_nsxt_security_group_test.go [hunk 7]
```diff
@@ -0,0 +1,477 @@
...
+	  end_address   = "212.1.${count.index}.20"
+	}
+}
+
+# Create stanadlone VM to check membership
+resource "vcd_vm" "emptyVM" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+
+    power_on      = false
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_security_group_test.go:237]: This script is nor well formatted

File: vcd/resource_vcd_nsxt_security_group_test.go [hunk 8]
```diff
@@ -0,0 +1,477 @@
...
+`
+
+const testAccNsxtSecurityGroup = testAccNsxtSecurityGroupPrereqs + `
+resource "vcd_nsxt_security_group" "group1" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  name = "test-security-group"
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_security_group_test.go:309]: This script is nor well formatted

File: vcd/resource_vcd_nsxt_security_group_test.go [hunk 9]
```diff
@@ -0,0 +1,477 @@
...
+	name = "{{.EdgeGw}}"
+}
+
+resource "vcd_nsxt_security_group" "group1" {
+	org  = "{{.Org}}"
+	vdc  = "{{.NsxtVdc}}"
+  
+	edge_gateway_id = data.vcd_edgegateway.existing.id
+  
+	name = "test-security-group"
```
Comment (dataclouder) [inline vcd/resource_vcd_nsxt_security_group_test.go:412]: This script is nor well formatted

File: website/docs/r/network_routed_v2.html.markdown [hunk 10]
```diff
@@ -40,6 +40,38 @@ resource "vcd_network_routed_v2" "nsxt-backed" {
...
 
+## Example Usage (NSX-T backed routed Org VDC network and a DHCP pool)
+
+```hcl
+resource "vcd_network_routed_v2" "parent-network" {
+  name = "nsxt-routed-dhcp"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
+
+  gateway = "7.1.1.1"
```
Comment (dataclouder) [inline website/docs/r/network_routed_v2.html.markdown:51]: This script is nor well formatted

File: website/docs/r/nsxt_security_group.html.markdown [hunk 11]
```diff
@@ -0,0 +1,97 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_nsxt_security_group"
+sidebar_current: "docs-vcd-resource-nsxt-security-group"
+description: |-
+Provides a resource to manage NSX-T Security Group. Security groups are groups of data center
```
Comment (dataclouder) [inline website/docs/r/nsxt_security_group.html.markdown:6]: Description text should be indented.
Screenshot 2021-05-04 at 14 54 56

File: website/docs/d/nsxt_security_group.html.markdown [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+page_title: "VMware Cloud Director: vcd_nsxt_security_group"
+sidebar_current: "docs-vcd-datasource-nsxt-security-group"
+description: |-
+  Provides a datasource to access NSX-T Security Group configuration. Security groups are groups of
+  data center group networks to which distributed firewall rules apply. Grouping networks helps you 
+  to reduce the total number of distributed firewall rules to be created. 
+---
+
+# vcd\_nsxt\_security\_group
+
```
Comment (dataclouder) [inline website/docs/d/nsxt_security_group.html.markdown:12]: Missing version availability

File: website/docs/r/nsxt_security_group.html.markdown [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+group networks to which distributed firewall rules apply. Grouping networks helps you to reduce
+the total number of distributed firewall rules to be created.
+---
+
+# vcd\_nsxt\_security\_group
+
```
Comment (dataclouder) [inline website/docs/r/nsxt_security_group.html.markdown:12]: Missing version availability

**Review by dataclouder — CHANGES_REQUESTED** [review]
Some small changes required

File: vcd/resource_vcd_nsxt_security_group.go (hunk 5 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_security_group.go]: Fixed

File: vcd/resource_vcd_nsxt_security_group_test.go (hunk 6 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_security_group_test.go]: Reformatted all scripts with `terraform fmt` in this file.

File: vcd/resource_vcd_nsxt_security_group_test.go (hunk 7 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_security_group_test.go]: Reformatted all scripts with `terraform fmt` in this file.

File: vcd/resource_vcd_nsxt_security_group_test.go (hunk 8 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_security_group_test.go]: Reformatted all scripts with `terraform fmt` in this file.

File: vcd/resource_vcd_nsxt_security_group_test.go (hunk 9 above)
Comment (Didainius) [reply vcd/resource_vcd_nsxt_security_group_test.go]: Reformatted all scripts with `terraform fmt` in this file.

File: website/docs/r/network_routed_v2.html.markdown (hunk 10 above)
Comment (Didainius) [reply website/docs/r/network_routed_v2.html.markdown:51]: Reformatted all scripts with `terraform fmt` in this file.

File: website/docs/r/nsxt_security_group.html.markdown (hunk 11 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown]: Shold be better now. My `make website`is broken, but this https://registry.terraform.io/tools/doc-preview looks to be showing good.

File: website/docs/d/nsxt_security_group.html.markdown (hunk 12 above)
Comment (Didainius) [reply website/docs/d/nsxt_security_group.html.markdown:12]: It was just a bit below, moved it up

File: website/docs/r/nsxt_security_group.html.markdown (hunk 13 above)
Comment (Didainius) [reply website/docs/r/nsxt_security_group.html.markdown:12]: It was just a bit below, moved it up

**Review by dataclouder — APPROVED** [review]
LGTM

**Pull request merged**